### PR TITLE
Convert registry to k8s.gcr.io

### PR DESCRIPTION
**Pull request by thockin**
This PR was auto-generated.  Please apply human expertise to review for correctness.

Followup to https://github.com/kubernetes/kubernetes/pull/54174 and https://github.com/kubernetes/kubernetes/pull/57824

xref https://github.com/kubernetes/release/issues/281

### CODE REVIEW

**Comment by thockin** [comment]
I am unfamiliar with the rules for version increments - does this really warrant that?

**Comment by mattfarina** [comment]
@thockin Charts are immutable. We want people to be able to check for things like a changed hash at a version. So, the version matters.

I'll take this one on and carry it home. Might take a couple weeks given the holidays but poke me if I don't get it done.

**Comment by mattfarina** [comment]
@thockin unless there's an objection, we're going to break this up into 12 separate pull requests. One for each chart so we can make sure all tests pass and the charts are immutable. We'll take care of it.

File: stable/mongodb-replicaset/init/Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 all: push
 
 TAG = 0.5
-PREFIX = gcr.io/google_containers/mongodb-install
+PREFIX = k8s.gcr.io/mongodb-install
```
Comment (dekkagaijin) [inline stable/mongodb-replicaset/init/Makefile]: `PREFIX = staging-k8s.gcr.io/mongodb-install`

**Review by dekkagaijin — CHANGES_REQUESTED** [review]

**Comment by thockin** [comment]
Since you're merging these piecemeal, I'll repurpose this to the next tweak.  Pushes go to one place, pulls from another.

**Comment by thockin** [comment]
@mattfarina

**Comment by mattfarina** [comment]
@foxish Any idea why these tests are failing on the mongodb-replicast chart?

**Comment by viglesiasce** [comment]
This test appears to be failing be cause our CI is not waiting until the RS is completely deployed, in the latest result only 2 of 3 replicas was up/running. Testing manually showed that this does indeed work.

**Comment by mattfarina** [comment]
@viglesiasce @unguiculus I think the issue is in how we test for statefulsets. When the first replica comes online all the pods are ready so it proceeds with the tests. If there are multiple replicas those start to come online but aren't ready. Testing shouldn't proceed until all the replicas are up. This is a bug in CI, I think.

**Comment by mattfarina** [comment]
/retest

A change to CI was merged that, I think, fixes the issue.

**Comment by thockin** [comment]
@mattfarina I see a handful of new uses of the old names.  Do youw ant me to send a PR or can you handle it?

**Pull request merged**